packages/vite/src/node/plugins/resolve.ts

```diff
@@ -410,59 +422,69 @@ export function resolvePlugin(
             importer,
             options,
             false,
+            this.environment.config,
             external,
           ))
         ) {
           return res
         }
 
         if (
-          (res = tryNodeResolve(id, importer, options, depsOptimizer, external))
+          (res = tryNodeResolve(
+            id,
+            importer,
+            options,
+            depsOptimizer,
+            external,
+            this.environment.config,
+          ))
         ) {
           return res
         }
 
-        // node built-ins.
-        // externalize if building for a node compatible environment, otherwise redirect to empty module
-        if (isBuiltin(id)) {
-          if (currentEnvironmentOptions.consumer === 'server') {
-            if (
-              options.enableBuiltinNoExternalCheck &&
-              options.noExternal === true &&
-              // if both noExternal and external are true, noExternal will take the higher priority and bundle it.
-              // only if the id is explicitly listed in external, we will externalize it and skip this error.
-              (options.external === true || !options.external.includes(id))
-            ) {
-              let message = `Cannot bundle Node.js built-in "${id}"`
-              if (importer) {
-                message += ` imported from "${path.relative(
-                  process.cwd(),
-                  importer,
-                )}"`
-              }
-              message += `. Consider disabling environments.${this.environment.name}.noExternal or remove the built-in dependency.`
-              this.error(message)
+        // built-ins
+        // externalize if building for a server environment, otherwise redirect to an empty module
+        if (
+          currentEnvironmentOptions.consumer === 'server' &&
+          isBuiltin(this.environment.config.resolve.builtins, id)
```

Comment:
Instead of passing down `this.environment.config` everywhere, you can use `options.builtins`.
It is merged here.
https://github.com/vitejs/vite/blob/8c0bc0a0624148b9d99924c6410f07a3ad25b5dd/packages/vite/src/node/plugins/resolve.ts#L236-L241

Reply:
oops, I see! sorry my bad! 😓